### PR TITLE
[yioremote] Make the Binding compatible to new firmware version of YIO Dock

### PR DESCRIPTION
**Pull request by miloit**
Sign Off Michael Loercher <michaelloercher@web.de>

Make the binding compatible to the new firmware

Make the binding compatible to the new firmware

Make the binding compatible to the new firmware

Tested with my setup

### CODE REVIEW

**Comment by miloit** [comment]
closed

File: bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-base-option-1.raw [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-IINST3 000 J
-IMAX1 026 8
-IMAX2 018 :
-IMAX3 027 ;
-PMAX 07990 ?
-PAPP 00540 *
-MOTDETAT 400000 F
-PPOT 00 #
-
+MOTDETAT 000000 B
```
Comment (cpmeister) [inline bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-base-option-1.raw:1]: Why is this file changed?

File: bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-ejp-option-1.raw [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
-IINST3 008 R
-IMAX1 038 ;
-IMAX2 042 7
-IMAX3 044 :
-PMAX 17480 :
-PAPP 05800 .
-MOTDETAT 000000 B
-PPOT 00 #
-
+MOTDETAT 000000 B
```
Comment (cpmeister) [inline bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-ejp-option-1.raw:1]: this one too

File: bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-base-option-1.raw (hunk 1 above)
Comment (miloit) [reply bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-base-option-1.raw:1]: Sorey don't know why.....how can I take it back?

File: bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-ejp-option-1.raw (hunk 2 above)
Comment (miloit) [reply bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-ejp-option-1.raw:1]: Sorry don't know why.....how can I take it back?

Comment (miloit) [reply bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-ejp-option-1.raw]: revert to origin

File: bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-base-option-1.raw (hunk 1 above)
Comment (miloit) [reply bundles/org.openhab.binding.teleinfo/src/test/resources/cbetm-base-option-1.raw]: revert to origin

**Review by miloit — COMMENTED** [review]
resolved

**Comment by miloit** [comment]
Why is the build failed? Is this an issue with my code?

**Comment by cpmeister** [comment]
The build failure is unrelated to your code, but the build also failed before it could attempt building the yioremote project. So I'll need to rerun the build again.

File: bundles/org.openhab.binding.yioremote/.classpath [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" output="target/classes" path="src/main/java">
```
Comment (cpmeister) [inline bundles/org.openhab.binding.yioremote/.classpath:3]: Please remove this file from this PR.

File: bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/dto/PingMessage.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        pingMessage.addProperty("type", type);
+        pingMessage.addProperty("command", command);
+        return pingMessage;
+    }
+
+    public String getPingMessageString() {
+        JsonObject pingMessage = new JsonObject();
+        pingMessage.addProperty("type", type);
+        pingMessage.addProperty("command", command);
+        return pingMessage.toString();
```
Comment (cpmeister) [inline bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/dto/PingMessage.java:53]:
```suggestion
        return getPingMessageJsonObject().toString();
```

File: bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java [hunk 5]
```diff
@@ -337,11 +337,30 @@ private void disposeWebsocketPollingJob() {
...
+            case SEND_PING:
+                if (resetHeartbeat()) {
+                    sendMessage(YioRemoteMessages.HEARTBEAT_MESSAGE, "");
+                    yioRemoteDockActualStatus = YioRemoteDockHandleStatus.CHECK_PONG;
+                } else {
+                    yioRemoteDockActualStatus = YioRemoteDockHandleStatus.COMMUNICATION_ERROR;
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Connection lost no ping from YIO DOCK");
+                    disposeWebsocketPollingJob();
+                    reconnectWebsocket();
```
Comment (cpmeister) [inline bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java:356]: `resetHeartbeat()` always returns true now, so this code will never get called anymore...

**Review by cpmeister — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.yioremote/.classpath (hunk 3 above)
Comment (miloit) [reply bundles/org.openhab.binding.yioremote/.classpath:3]: How can I do this???

File: bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/dto/PingMessage.java (hunk 4 above)
Comment (miloit) [reply bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/dto/PingMessage.java:53]: done

File: bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java (hunk 5 above)
Comment (miloit) [reply bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java:356]: done

File: bundles/org.openhab.binding.yioremote/.classpath (hunk 3 above)
Comment (miloit) [reply bundles/org.openhab.binding.yioremote/.classpath]: removed

**Comment by miloit** [comment]
Is it an issue of my code that this is not build?

**Comment by cpmeister** [comment]
> Is it an issue of my code that this is not build?

Looks like it. I think you need to run `mvn spotless:apply`.
```
[ERROR] Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:2.0.3:check (codestyle_check) on project org.openhab.binding.yioremote: The following files had format violations:
[ERROR] src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java
[ERROR] @@ -342,7 +342,7 @@
[ERROR] ················yioRemoteDockActualStatus·=·YioRemoteDockHandleStatus.CHECK_PONG;
[ERROR] ················break;
[ERROR] ············case·SEND_PING:
[ERROR] -\t\t\t\tresetHeartbeat();
[ERROR] +················resetHeartbeat();
[ERROR] ················sendMessage(YioRemoteMessages.HEARTBEAT_MESSAGE,·"");
[ERROR] ················yioRemoteDockActualStatus·=·YioRemoteDockHandleStatus.CHECK_PONG;
[ERROR] ················break;
[ERROR] Run 'mvn spotless:apply' to fix these violations.
```

**Review by cpmeister — APPROVED** [review]
LGTM
